frontend/packages/shared/src/components/GiteaHeader/ThreeDotsMenu/ThreeDotsMenu.test.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { render as rtlRender, act, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { ThreeDotsMenu, ThreeDotsMenuProps } from './ThreeDotsMenu';
+import { textMock } from '../../../../../../testing/mocks/i18nMock';
+
+const user = userEvent.setup();
+const cloneTextMock = textMock('sync_header.clone');
+const repositoryTextMock = textMock('dashboard.repository');
+const localChangesTextMock = textMock('dashboard.local_changes');
+
+jest.mock(
+  'app-shared/components/GiteaHeader/ThreeDotsMenu/LocalChangesModal/LocalChangesModal',
+  () => ({
+    LocalChangesModal: jest.fn().mockReturnValue(null),
+  }),
+);
+
+describe('ThreeDotsMenu', () => {
+  afterEach(jest.clearAllMocks);
+
+  it('should show the menu items when open', async () => {
+    await render();
+    const threeDotsMenu = screen.getByLabelText('Gitea menu');
+    expect(threeDotsMenu).toBeInTheDocument();
+
+    await act(() => user.click(threeDotsMenu));
+    const cloneText = await screen.findByText(cloneTextMock);
+    expect(cloneText).toBeInTheDocument();
+
+    const repoText = await screen.findByText(repositoryTextMock);
+    expect(repoText).toBeInTheDocument();
+
+    const localchangeText = await screen.findByText(localChangesTextMock);
+    expect(localchangeText).toBeInTheDocument();
+  });
+
+  it('should not show the clone option when onlyShowRepository is true', async () => {
+    await render({ onlyShowRepository: true });
+
+    const threeDotsMenu = screen.getByLabelText('Gitea menu');
+    expect(threeDotsMenu).toBeInTheDocument();
+
+    await act(() => user.click(threeDotsMenu));
+
+    const cloneText = screen.queryByText(cloneTextMock);
+    expect(cloneText).not.toBeInTheDocument();
+  });
+
+  it('should local changes be in document when setLocalChangesModalIsOpen and it is clicked', async () => {
```

Comment:
I dont think this test does what it suppose to do 🤔 
You will need some sort of test-id in the mocked local changes modal for this to make sense. Or dont mock the modal at all and look for the actual text rendered inside it.

The test runs because we are looking for the text in the threeDotsMenu both before and after the 'click'.

Reply:
I changed it just to render local changes.